src/compiler/crystal/codegen/exception.cr

```diff
@@ -195,6 +203,153 @@ class Crystal::CodeGenVisitor
     false
   end
 
+  private def codegen_windows_seh(node : ExceptionHandler)
```

Comment:
1. I think there should be some comments here explaining how the exception data is handled. I know it comes from documentation of llvm seh and trial an error. But there are some decisions w.r.t. how exception is encoded as `void*` that would help future readers. Also some basic examples of the result in `llvm-ir` would be great for documentation IMO (l tried to do that at for [overflow here](https://github.com/bcardiff/crystal/blob/5ff7481718dd8fe8a6260d30d9647cffd8076d16/src/compiler/crystal/codegen/primitives.cr#L263-L291) in the past).

2. And I am tempted to suggest some comment or extraction of common parts with the codegen_landing_pad. The handling of multiple ensures is shared. Maybe a strategy struct can be used to encapsulate custom logic for pads vs seh, leaving the existing comments `visit(node : ExceptionHandler)`/`codegen_landing_pad` that goes into the details of the handling of the AST. The only real value of this would be to help the reader to understand separate the concepts that are together in this code.